src/lib/y2users/collision_checker.rb

```diff
@@ -0,0 +1,127 @@
+# find current contact information at www.suse.com.
+
+require "y2issues"
+
+module Y2Users
+  # Internal class to check if there are collisions in config.
+  # This is not part of the stable API.
+  class CollisionChecker
+    include Yast::I18n
+
+    # Issue location describing the Group#gid attribute
+    GROUP_ID_LOC = "group:gid".freeze
+    # Issue location describing the Group#name attribute
+    GROUP_NAME_LOC = "group:name".freeze
+    # Issue location describing the User#uid attribute
+    USER_ID_LOC = "user:uid".freeze
+    # Issue location describing the User#uid attribute
+    USER_NAME_LOC = "user:name".freeze
+    private_constant :GROUP_ID_LOC, :GROUP_NAME_LOC, :USER_ID_LOC, :USER_NAME_LOC
+
+    # Constructor
+    #
+    # @param config [Y2Users::Config] config to detect colliding ids
+    def initialize(config)
+      textdomain "users"
+      @config = config
+    end
+
+    # Returns a list of issues found while checking collisions
+    #
+    # @return [Y2Issues::List]
+    def issues
+      list = Y2Issues::List.new
+
+      list.concat(duplicite_users_ids)
+      list.concat(duplicite_users_names)
+      list.concat(duplicite_groups_ids)
+      list.concat(duplicite_groups_names)
+
+      list
+    end
+
+  private
+
+    # @return [Y2Users::Config] config to validate
+    attr_reader :config
+
+    def duplicite_users_ids
+      grouped = config.users.all.group_by(&:uid)
+      grouped.delete(nil)
+      grouped.select! { |_uid, users| users.size > 1 } # colliding uids
+
+      issues = grouped.map do |uid, users|
+        msg = format(_("Users %{users} have same UID %{uid}."),
+          users: users.map(&:name).join(", "),
+          uid:   uid)
+        Y2Issues::Issue.new(msg, location: USER_ID_LOC, severity: :warn)
+      end
+
+      Y2Issues::List.new(issues)
+    end
+
+    def duplicite_users_names
+      grouped = config.users.all.group_by(&:name)
+      # missing or invalid name is caught by user validation, so skip it here
+      grouped.delete(nil)
+      grouped.delete("")
+      grouped.select! { |_name, users| users.size > 1 } # colliding uids
+
+      issues = grouped.map do |name, _users|
+        msg = format(_("User %{user} is specified multiple times."),
+          user: name)
+        Y2Issues::Issue.new(msg, location: USER_NAME_LOC, severity: :warn)
+      end
+
+      Y2Issues::List.new(issues)
+    end
+
+    def duplicite_groups_ids
+      grouped = config.groups.all.group_by(&:gid)
+      grouped.delete(nil)
+      grouped.select! { |_gid, groups| groups.size > 1 } # colliding uids
+
+      issues = grouped.map do |gid, groups|
+        msg = format(_("Groups %{groups} have same GID %{gid}."),
+          groups: groups.map(&:name).join(", "),
+          gid:    gid)
+        Y2Issues::Issue.new(msg, location: GROUP_ID_LOC, severity: :warn)
+      end
+
+      Y2Issues::List.new(issues)
+    end
+
+    def duplicite_groups_names
```

Comment:
NP: don't you feel the code of the four methods is too similar? Cannot the logic be extracted to some base method?